sql/core/src/main/scala/org/apache/spark/sql/execution/limit.scala

```diff
@@ -83,9 +83,8 @@ trait BaseLimitExec extends UnaryExecNode with CodegenSupport {
     s"""
        | if ($countTerm < $limit) {
        |   $countTerm += 1;
+       |   if ($countTerm == $limit) $stopEarly = true;
```

Comment:
Set stop early flag to true so we won't step into next element in iterator. Otherwise, we still go to get next element from the iterator. So if the limitation is 1, we will get 2 elements from the iterator.

Note: This won't cause real problem because we have an `if` to guard against processing element.